### PR TITLE
Use godwoken-docker-prebuilds image in docker/docker-compose.yml

### PR DESCRIPTION
**Pull request by Flouse**
- [x] refactor `make init` step 
   - [x] remove `cargo build` step
   - [x] remove init containers
   - [x] move `yarn install` to init stage
- [x] instead ckb container 
- [x] instead godwoken container
   - [x] move godwoken_install_entrypoint.sh into `make init`
- [x] instead polyjuice container
- [x] instead web3 container
- [x] web3 container need `jq` 
   - [x]  https://github.com/RetricSu/godwoken-kicker/blob/master/web3_entrypoint.sh#L5

### CODE REVIEW

**Comment by RetricSu** [comment]
> I have used `make install` to update all the submodules successfully.
> But still, I got the signature error.
> @RetricSu Would you please help me to check it?
> 
> https://github.com/Flouse/godwoken-kicker
> 
> ```
> godwoken_1             | [INFO] 04:11:25 ts-node-dev ver. 1.1.6 (using ts-node ver. 9.1.1, typescript ver. 4.1.3)
> godwoken_1             | Error: /code/godwoken-examples/node_modules/@ckb-lumos/indexer/native/index.node: undefined symbol: _ZN2v816FunctionTemplate3NewEPNS_7IsolateEPFvRKNS_20FunctionCallbackInfoINS_5ValueEEEENS_5LocalIS4_EENSA_INS_9SignatureEEEiNS_19ConstructorBehaviorENS_14SideEffectTypeE
> godwoken_1             |     at Module._extensions..node (internal/modules/cjs/loader.js:1122:18)
> godwoken_1             |     at Object.nodeDevHook [as .node] (/code/godwoken-examples/node_modules/ts-node-dev/lib/hook.js:63:13)
> godwoken_1             |     at Module.load (internal/modules/cjs/loader.js:928:32)
> godwoken_1             |     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
> godwoken_1             |     at Module.require (internal/modules/cjs/loader.js:952:19)
> godwoken_1             |     at require (internal/modules/cjs/helpers.js:88:18)
> godwoken_1             |     at Object.<anonymous> (/code/godwoken-examples/node_modules/@ckb-lumos/indexer/lib/index.js:4:59)
> godwoken_1             |     at Module._compile (internal/modules/cjs/loader.js:1063:30)
> godwoken_1             |     at Module._compile (/code/godwoken-examples/node_modules/source-map-support/source-map-support.js:547:25)
> godwoken_1             |     at Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
> godwoken_1             | [ERROR] 04:11:34 Error: /code/godwoken-examples/node_modules/@ckb-lumos/indexer/native/index.node: undefined symbol: _ZN2v816FunctionTemplate3NewEPNS_7IsolateEPFvRKNS_20FunctionCallbackInfoINS_5ValueEEEENS_5LocalIS4_EENSA_INS_9SignatureEEEiNS_19ConstructorBehaviorENS_14SideEffectTypeE
> ```

do not recognise this error. need more info, like what trigger this error. deposit? create creator? deploy contract?

**Comment by Flouse** [comment]
> https://github.com/Flouse/godwoken-kicker
> 
> ```
> godwoken_1             | [INFO] 04:11:25 ts-node-dev ver. 1.1.6 (using ts-node ver. 9.1.1, typescript ver. 4.1.3)
> godwoken_1             | Error: /code/godwoken-examples/node_modules/@ckb-lumos/indexer/native/index.node: undefined symbol: _ZN2v816FunctionTemplate3NewEPNS_7IsolateEPFvRKNS_20FunctionCallbackInfoINS_5ValueEEEENS_5LocalIS4_EENSA_INS_9SignatureEEEiNS_19ConstructorBehaviorENS_14SideEffectTypeE
> godwoken_1             |     at Module._extensions..node (internal/modules/cjs/loader.js:1122:18)
> godwoken_1             |     at Object.nodeDevHook [as .node] (/code/godwoken-examples/node_modules/ts-node-dev/lib/hook.js:63:13)
> godwoken_1             |     at Module.load (internal/modules/cjs/loader.js:928:32)
> godwoken_1             |     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
> godwoken_1             |     at Module.require (internal/modules/cjs/loader.js:952:19)
> godwoken_1             |     at require (internal/modules/cjs/helpers.js:88:18)
> godwoken_1             |     at Object.<anonymous> (/code/godwoken-examples/node_modules/@ckb-lumos/indexer/lib/index.js:4:59)
> godwoken_1             |     at Module._compile (internal/modules/cjs/loader.js:1063:30)
> godwoken_1             |     at Module._compile (/code/godwoken-examples/node_modules/source-map-support/source-map-support.js:547:25)
> godwoken_1             |     at Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
> godwoken_1             | [ERROR] 04:11:34 Error: /code/godwoken-examples/node_modules/@ckb-lumos/indexer/native/index.node: undefined symbol: _ZN2v816FunctionTemplate3NewEPNS_7IsolateEPFvRKNS_20FunctionCallbackInfoINS_5ValueEEEENS_5LocalIS4_EENSA_INS_9SignatureEEEiNS_19ConstructorBehaviorENS_14SideEffectTypeE
> ```

```
  godwoken:
    image: nervos/godwoken-prebuilds:v0.2.0-rc2
```
The node version in nervos/godwoken-prebuilds:v0.2.0-rc2 is 14, while the previous node version is 12.
So we should reinstall the dependencies totally, and the above error was fixed.

File: web3_entrypoint.sh [hunk 1]
```diff
@@ -47,5 +45,7 @@ PORT=8024
 CREATOR_ACCOUNT_ID=3
 EOF
 
+# pwd => /web3
+yarn install
```
Comment (RetricSu) [inline web3_entrypoint.sh:49]: and this one too

File: godwoken_entrypoint.sh [hunk 2]
```diff
@@ -17,6 +17,7 @@ source ${PROJECT_DIR}/gw_util.sh
 
 # wait for polyjuice complete preparing money before godwoken deployment, avoiding cell comsuming conflict.
 cd $PolyjuiceDir
+yarn install
```
[the remark this answers is not in the document]
Comment (Flouse) [reply godwoken_entrypoint.sh:20]: https://github.com/RetricSu/godwoken-examples/pull/8

**Comment by Flouse** [comment]
Error: this.indexer not found.

$ yarn workspace @godwoken-examples/runner prepare-money
$ LUMOS_CONFIG_FILE=./configs/lumos-config.json MODE=docker-compose ts-node-dev ./src/prepare_money.ts
[INFO] 10:09:12 ts-node-dev ver. 1.1.6 (using ts-node ver. 9.1.1, typescript ver. 4.2.4)
start..
Error: this.indexer not found.
    at Api.checkCkbBalance (/godwoken-examples/packages/runner/src/api.ts:1137:13)
    at Object.run (/godwoken-examples/packages/runner/src/prepare_money.ts:25:36)
    at Object.<anonymous> (/godwoken-examples/packages/runner/src/prepare_money.ts:50:1)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Module._compile (/godwoken-examples/node_modules/source-map-support/source-map-support.js:547:25)
    at Module.m._compile (/tmp/ts-node-dev-hook-15625269222695226.js:69:33)
    at Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at require.extensions..jsx.require.extensions..js (/tmp/ts-node-dev-hook-15625269222695226.js:114:20)
    at require.extensions.<computed> (/tmp/ts-node-dev-hook-15625269222695226.js:71:20)
    at Object.nodeDevHook [as .ts] (/godwoken-examples/node_modules/ts-node-dev/lib/hook.js:63:13)
(node:96) UnhandledPromiseRejectionWarning: Error: internal error in Neon module: Failed to open rocksdb: Error { message: "Failed to create RocksDB directory: `Os { code: 13, kind: PermissionDenied, message: \"Permission denied\" }`." }

**Comment by RetricSu** [comment]
> Error: this.indexer not found.
> 
> $ yarn workspace @godwoken-examples/runner prepare-money
> $ LUMOS_CONFIG_FILE=./configs/lumos-config.json MODE=docker-compose ts-node-dev ./src/prepare_money.ts
> [INFO] 10:09:12 ts-node-dev ver. 1.1.6 (using ts-node ver. 9.1.1, typescript ver. 4.2.4)
> start..
> Error: this.indexer not found.
> at Api.checkCkbBalance (/godwoken-examples/packages/runner/src/api.ts:1137:13)
> at Object.run (/godwoken-examples/packages/runner/src/prepare_money.ts:25:36)
> at Object. (/godwoken-examples/packages/runner/src/prepare_money.ts:50:1)
> at Module._compile (internal/modules/cjs/loader.js:1063:30)
> at Module._compile (/godwoken-examples/node_modules/source-map-support/source-map-support.js:547:25)
> at Module.m._compile (/tmp/ts-node-dev-hook-15625269222695226.js:69:33)
> at Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
> at require.extensions..jsx.require.extensions..js (/tmp/ts-node-dev-hook-15625269222695226.js:114:20)
> at require.extensions. (/tmp/ts-node-dev-hook-15625269222695226.js:71:20)
> at Object.nodeDevHook [as .ts] (/godwoken-examples/node_modules/ts-node-dev/lib/hook.js:63:13)
> (node:96) UnhandledPromiseRejectionWarning: Error: internal error in Neon module: Failed to open rocksdb: Error { message: "Failed to create RocksDB directory: `Os { code: 13, kind: PermissionDenied, message: \"Permission denied\" }`." }

this usually happen when we have two indexer running using the same indexer-data-folder. indexer-data-folder is like the following path. https://github.com/RetricSu/godwoken-examples/blob/57d10ff389794fdc38aabe666408053afa39dd15/packages/runner/src/prepare_money.ts#L7
did you happen to be running two instance of api comsuming the same data-folder ?

**Comment by Flouse** [comment]
it was caused by privilege less.
https://github.com/Flouse/godwoken-examples/blob/d6a89ebb4793863845b9be707618801faa7e3e30/Dockerfile

File: web3_entrypoint.sh (hunk 1 above)
Comment (Flouse) [reply web3_entrypoint.sh]: Done -> https://github.com/Flouse/godwoken-web3/blob/docker/Dockerfile

**Comment by Flouse** [comment]
Tests：

- [x] make clean -> make init -> make start 
- [ ] make stop -> make start
- [x] make start-f

**Pull request merged**